modules/internal/ChapelBase.chpl

```diff
@@ -1488,6 +1488,52 @@ module ChapelBase {
       throw new owned TaskErrors(e.errors);
   }
 
+  // Routines for re-interpreting reals as uints and vice-versa, at
+  // the bit level
+  //
+  @unstable "This routine may change names / signatures"
+  proc param (real(64)).toBits() param : uint {
+    param ui: uint(64) = __primitive("real64 as uint64", this);
+    return ui;
+  }
+
+  @unstable "This routine may change names / signatures"
+  proc param (real(32)).toBits() param : uint(32) {
+    param ui: uint(32) = __primitive("real32 as uint32", this);
+    return ui;
+  }
+
+  @unstable "This routine may change names / signatures"
+  inline proc (real(?w)).toBits(): uint(w) {
```

Comment:
Why is this one generic on `?w` while the param versions are concrete?

Reply:
To make the param versions generic, I would've needed to add a conditional to their bodies to select one of two primitives, and the approach of just having two calls seemed simpler given how short they were.  In this case, nothing in the body needed to vary based on the type, so it seemed preferable to make it generic.